static/js/hotkey.js

```diff
@@ -496,6 +496,11 @@ export function process_tab_key() {
         return emoji_picker.navigate("tab");
     }
 
+    if ($("#compose-send-button").is(":focus")) {
+        $(".compose_upload_file").trigger("focus");
+        return true;
+    }
```

Comment:
This will cause problems if uploading files is disabled and thus that button doesn't appear.  I haven't read the rest of the branch for the motivation, but this is a reason to have the HTML ordered so that tabindex will do the right thing automatically.

Reply:
I think if make this trigger a focus on the first element inside `compose_control_buttons_container`, that'd resolve this bug in a reasonably clean way?  Though we should check that `Shift+Tab` takes you back properly too.

Reply:
Fixed it by using the first element. `Shift+tab` takes you to compose text area, which is fine?

I am not sure that it is worth investing too much in `tab+enter` sends meta. It is non-intuitive for a new Zulip user even if the user is a proper keyboard user. The user is more likely to use `CMD/Ctrl + Enter` to send before trying to `tab` to the send button.

Reply:
Having the tab-based keyboard navigation is important: when we've had even minor bugs in this logic we've gotten some very grumpy complaints.  I expect this change, which makes `Shift+Tab` not behave like normal browser elements will likely result in some people unhappy.

Having skimmed the CSS for this, I'm not sure I understand what benefit we get from reordering the elements to not have the send button first in the HTML, as this PR does?  We seem to still have `float: right` CSS present.  But I haven't tried to understand it deeply.

Reply:
Okay, I dropped this commit. There was `tabindex` bug due to which `tab` from `send` to `file upload` icon wasn't working naturally with the send button element ordered above it in DOM.